### PR TITLE
Tabular Connectors - Rework SwaggerTabularConnector and various fixes

### PR DESCRIPTION
**Pull request by LucGenetier**
Rework SwaggerTabularConnector function identification logic
Remove all hardcoded names (table, default...)
Rename DataSetName to DatasetName
Add GetGlobalValueNames which returns the list of expected global value names
Move OpenApiDocument to constructor
Better detection of missing global values
CdpTabularConnetor: add Name and DisplayName (only valid after Init)
ConnectorType: Fix DisplayName and add Summary
Add SalesForce tests

### CODE REVIEW

**Comment by LucGenetier** [comment]
Public API changes: <br />❌CP0002 M:Microsoft.PowerFx.Connectors.CdpTabularService.get_DataSetName<br />❌CP0002 M:Microsoft.PowerFx.Connectors.SwaggerTabularService.#ctor(System.Collections.Generic.IReadOnlyDictionary{System.String,Microsoft.PowerFx.Types.FormulaValue})

**Comment by LucGenetier** [comment]
Public API changes: <br />❌CP0002 M:Microsoft.PowerFx.Connectors.CdpTabularService.get_DataSetName<br />❌CP0002 M:Microsoft.PowerFx.Connectors.SwaggerTabularService.#ctor(System.Collections.Generic.IReadOnlyDictionary{System.String,Microsoft.PowerFx.Types.FormulaValue})

File: src/libraries/Microsoft.PowerFx.Connectors/Environment/CdpTabularService.cs [hunk 1]
```diff
@@ -16,19 +16,27 @@ namespace Microsoft.PowerFx.Connectors
     // Implements CDP protocol for Tabular connectors
     public class CdpTabularService : TabularService
     {
-        public string DataSetName { get; }
+        public string Name { get; private set; }
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Connectors/Environment/CdpTabularService.cs:19]: >Name [](http://example.com/codeflow?start=22&length=4)

Name is too ambigious... 

Can we comment difference between these names (and maybe with a common example of each)

**Review by MikeStall — APPROVED** [review]
:shipit:

File: src/libraries/Microsoft.PowerFx.Connectors/Environment/CdpTabularService.cs (hunk 1 above)
Comment (LucGenetier) [reply src/libraries/Microsoft.PowerFx.Connectors/Environment/CdpTabularService.cs:19]: Will do in future PR

**Pull request merged**